### PR TITLE
raise GokartbuildError on gokart.build failed

### PR DESCRIPTION
**Pull request by snowhork**
This MR for https://github.com/m3dev/gokart/issues/209, https://github.com/m3dev/gokart/issues/229

### CODE REVIEW

**Comment by snowhork** [comment]
@vaaaaanquish @Hi-king @hirosassa please review.

File: test/test_build.py [hunk 1]
```diff
@@ -65,6 +72,10 @@ def test_build_dict_outputs(self):
         output = gokart.build(_DummyTaskTwoOutputs(param1=param_dict['out1'], param2=param_dict['out2']), reset_register=False)
         self.assertEqual(output, param_dict)
 
+    def test_failed_task(self):
+        with self.assertRaises(GokartBuildError):
+            gokart.build(_DummyFailedTask(), reset_register=False)
+
```
Comment (hirosassa) [inline test/test_build.py:78]: nice test!

**Review by hirosassa — APPROVED** [review]
Thanks! LGTM!

**Comment by vaaaaanquish** [comment]
LGTM. plz update documents.

**Comment by snowhork** [comment]
@vaaaaanquish Thanks for your review. I've added docs. please check.

**Review by vaaaaanquish — APPROVED** [review]

**Comment by hirosassa** [comment]
@snowhork Thanks for your great contribution! merging

**Pull request merged**